db/src/urls.rs

```diff
@@ -166,7 +166,11 @@ pub async fn generate_and_store_summary(
     Ok(())
 }
 
-pub async fn persist_article(pool: &PgPool, page: Page, markdown: Markdown) -> color_eyre::Result<PageSnapShot> {
+pub async fn persist_article(
```

Comment:
This ensures that the HTML is properly formatted and the URL is valid before being saved as a PageSnapShot.